### DISCUSSION TITLE
[help]内容集合的图片发布到线上无法展示的404

### DISCUSSION THREAD

**Discussion by lkzwc** [withastro/docs#7264] [Ideas]
比如 MDX中写的是

---
cover: "/src/content/hotcard/_image/cangjun.jpg"
---

最终发布到线上环境 访问的图片就是 www.domain.com/src/content/hotcard/_image/cangjun.jpg 结果就挂掉了 
需要自己讲图片路径转化下吗 有一些具体的方案吗

**Comment by liruifengv** [comment]
你好。你应该用相对路径“./_image/canghun.jpg” 就会被自动处理了。
用/开头是网站根目录，只有存在public下的图片不会被转换，可以用/开头。

详情可以查看一下图片相关文档： 

https://docs.astro.build/zh-cn/guides/images/#%E5%86%85%E5%AE%B9%E9%9B%86%E5%90%88%E4%B8%AD%E7%9A%84%E5%9B%BE%E5%83%8F

**Reply by lkzwc** [reply]
`Image`'s and `getImage`'s `src` parameter must be an imported image or an URL, it cannot be a string filepath. Received `./_image/cangjun.jpg`.  如果换成绝对路径就直接报错了 


这个是schema的配置
 const hot = defineCollection({
  type: 'content',
    schema: ({ image })=>z.object({
      title: z.string(),
      pubDate: z.string(),
      description: z.string(),
      cover:  z.string().optional(),
      coverAlt: z.string(),
      author: z.string(),
      image: z.object({
        url:  z.string().optional(),
        alt: z.string()
      }),
      tags: z.array(z.string())
    })
 });

而且我看了下官方文档也配置的是绝对路径 https://github.com/withastro/astro.build/blob/main/src/content/blog/astro-440.mdx

**Reply by liruifengv** [reply]
不要看官网的例子，他这个做了处理的 https://github.com/withastro/astro.build/blob/main/src/content/blog/_resolveImage.ts

schema 更新一下：
```
const blogCollection = defineCollection({
  schema: ({ image }) => z.object({
    cover: image().refine((img) => img.width >= 1080, {
      message: "封面图片必须至少 1080 像素宽！",
    }).
     or(z.string())
    .optional(),
  }),
});
```

**Reply by lkzwc** [reply]
整个其他流程是OK的 但是build的时候 依旧就挂了

**Reply by lkzwc** [reply]
LocalImageUsedWrongly] `Image`'s and `getImage`'s `src` parameter must be an imported image or an URL, it cannot be a string filepath. Received `./_image/5g.webp`.

**Reply by liruifengv** [reply]
可以给一个最小复现链接吗

**Reply by lkzwc** [reply]
如果我是用了绝对链接 cover: "/src/content/hotcard/_image/cangjun.jpg" 这种情况下是OK的 dev 和 pre都是好的 但是pre内容界面就挂了 即博客内容页面图片不展示 

如果使用相对链接cover: "./_image/cangjun.jpg"  dev环境的内容页面就挂了


https://github.com/lkzwc/liuketh

我把仓库开放下 您看下 谢谢大佬🙏

**Reply by lkzwc** [reply]
目前已发布的版本是绝对全路径 但是 md 内容页面图片挂了

**Reply by liruifengv** [reply]
我给你提了个PR，应该解决了。
https://github.com/lkzwc/liuketh/pull/1

目前看来是在frontmatter中指定layout 和 Image 结合使用的问题，似乎是个bug，我提个 issue 跟进一下。

推荐使用导入layout的方式，自主可控：https://docs.astro.build/zh-cn/basics/layouts/#手动导入布局mdx

**Reply by lkzwc** [reply]
这尽然式布局惹的祸 万万没想到呀  感谢大佬 谢谢@liruifengv
